Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs

```diff
@@ -5,35 +5,141 @@
+            Console.WriteLine("Witamy w notatniku. Wybierz:");
+            Console.WriteLine("1 - nowa notatka");
+            Console.WriteLine("2 - otwórz notatkę");
+        }
 
-            Console.WriteLine("Witamy w notatniku. Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+        private void MenageUserInput()
+        {
+            string insertValue;
+            do
+            {
+                insertValue = Console.ReadLine();
 
-            note = Console.ReadLine();
+            } while (!IsUserInputValid(insertValue));
 
-            ConsoleKeyInfo typedKey = Console.ReadKey();
-            if (typedKey.Key == ConsoleKey.Enter)
+            GoToUserChoice(insertValue);
+        }
+
+        private bool IsUserInputValid(string userInput)
+        {
+            if (userInput == "1" || userInput == "2")
+            {
+                return true;
+            }
+
+            Console.WriteLine("Proszę wybrać numer z listy");
+            return false;
+        }
+
+        private void GoToUserChoice(string userChoice)
+        {
+            switch (userChoice)
             {
-                Console.WriteLine("To jest twoja notatka, która została zapisana do pliku txt: " + note);
-                SaveFile(note);
-                
+                case "1":
+                    WriteNewNote();
+                    break;
+                case "2":
+                    MenageUserPathInput();
+                    break;
+                default:
+                    Console.WriteLine("Błąd");
+                    break;
+            }
+        }
+
+        private void WriteNewNote()
+        {       
+            Console.WriteLine("Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+            string note = Console.ReadLine();
+            ConsoleKeyInfo typedKey = Console.ReadKey();
+
+            if (typedKey.Key == ConsoleKey.Enter && SaveFile(note))
+            {                
+                Console.WriteLine("Zapisano.");                                
                 var mainMenu = new Notatnik_parser.MainMenu.MainMenu();
                 mainMenu.Run();
-            }            
-        }      
+            }
+            else if (typedKey.Key == ConsoleKey.Enter && !SaveFile(note))
+            {                
+                Console.WriteLine("Błędna ścieżka. Nie udało się zapisać.");                                
+            }
+        }
+
+        private bool SaveFile(string textNoteParameter)
+        {    
+            Console.WriteLine(pathInstructions, "zapisać do");
+            string pathAndFileName = Console.ReadLine();
+
+            try
+            {
+                using (FileStream fs = File.Create(pathAndFileName))
+                {
+                    Byte[] textNote = new UTF8Encoding(true).GetBytes(textNoteParameter);
+                    fs.Write(textNote, 0, textNote.Length);
+                }
+            }
+            catch (UnauthorizedAccessException)
```

Comment:
czemu łapiesz ten typ wyjątku?

Reply:
The story is simple... I got this error in IDE during tests, I mentioned that I thoroughly tested this code. Sometimes happens that you try to save something somewhere where you aren't allowed to save this.